audio_extraction/audio_extraction.ipynb

```diff
@@ -0,0 +1,245 @@
+   "source": [
+    "def extract_audio_from_video(video_path, to_folder = None,\n",
+    "                             video_extensions = ['.mp4', '.mov'],\n",
+    "                             log_flag = False):\n",
+    "    \"\"\"\n",
+    "    Function that goes into the directory and extracts audio from all mp4 files and outputs it\n",
+    "    into a folder called 'extracted_audio'\n",
+    "    \n",
+    "    Arguments:\n",
+    "    video_path | pathlib.PosixPath\n",
+    "    Path to the directory that contains all of the video files.\n",
+    "    \n",
+    "    Optional Arguments:\n",
+    "    to_folder | pathlib.PosixPath\n",
+    "    Path to the directory where you want to store the extracted audio. By default,\n",
+    "    it will be extracted into a folder in video_path's parent directory as 'extracted_audio'\n",
+    "    \n",
+    "    video_extensions | list\n",
+    "    A list of strings that contains the extensions you want ffmpeg to extract audio from. The\n",
+    "    default includes .mp4 files and .mov files.\n",
+    "    \n",
+    "    log_flag | bool\n",
+    "    A boolean value that sets the logging level. If true, it will write the debug output to\n",
+    "    the log file `extract_audio_from_video.log`. Otherwise, it will only write warning and errors.\n",
+    "    \n",
+    "    Returns:\n",
+    "    It returns a string that tells you how many audio files were extracted.\n",
+    "    \"\"\"\n",
+    "    # Set up the logging for this function\n",
+    "    # filemode = 'w' makes it overwrite the previous log\n",
+    "    \n",
+    "    # Using dictionary as switch case for using whether or not to log the logging info\n",
+    "    log_switch_case = {\n",
+    "        # The only logging functions that are used are logging.info()\n",
+    "        True: logging.DEBUG, # This writes logging debug output into the log file\n",
+    "        False: logging.WARNING # This stops writing logging debug output into the log file\n",
+    "    }\n",
+    "    # See https://docs.python.org/3/howto/logging.html for descriptions of the various levels\n",
+    "    \n",
+    "    # Set up the logging\n",
+    "    logging.basicConfig(filename = 'extract_audio_from_video.log',\n",
+    "                    encoding = 'utf-8', level = log_switch_case[log_flag],\n",
+    "                    filemode = 'w')\n",
+    "    \n",
+    "    # Making a default directory for to_folder\n",
+    "    if to_folder == None:\n",
+    "        print(\"Created a default to_folder as none was given.\")\n",
+    "        video_path.parent.joinpath('extracted_audio')\n",
+    "    \n",
+    "    # Create a counter for the number of audio files processed\n",
+    "    audio_file_counter = 0\n",
+    "    \n",
+    "    # Go through all files in `video_path`\n",
+    "    for child in video_path.iterdir():\n",
+    "        # Debugging: show the file name\n",
+    "        logging.debug('Detected file: {}'.format(child.name))\n",
```

Comment:
using f-strings are a little more "modern" and cause less concat issues than .format, not worth changing code over, but for future reference

Reply:
I've edited the code to use f-strings and checked that the logs are still the same afterwards. Here's an example:
```
# Debugging: print the codec type
logging.debug(f'Codec type: {stream["codec_type"]}')
```

Thanks for the tip!